api/utils/slices.go

```diff
@@ -51,11 +51,27 @@ func Deduplicate(in []string) []string {
 		return in
 	}
 	out := make([]string, 0, len(in))
-	seen := make(map[string]bool, len(in))
+	seen := make(map[string]struct{}, len(in))
 	for _, val := range in {
 		if _, ok := seen[val]; !ok {
 			out = append(out, val)
-			seen[val] = true
+			seen[val] = struct{}{}
+		}
+	}
+	return out
+}
+
+// DeduplicateBytes deduplicates list of bytes
+func DeduplicateBytes(in [][]byte) [][]byte {
```

Comment:
Can we please add some test for this new function?

Reply:
Alternatively, consider generifying `utils.Deduplicate` since it's basically identical.